analyzers/tests/SonarAnalyzer.UnitTest/TestCases/ConditionEvaluatesToConstant.CSharp7.cs

```diff
@@ -334,4 +334,48 @@ public int Foo()
             return 1;
         }
     }
+
+    // https://github.com/SonarSource/sonar-dotnet/issues/3910
+    public class Repro_3910
+    {
+        public void ElementAccess1(Exception ex)
+        {
+            if (ex?.Data["Key"] is string value)
+            {
+            }
+            else if (ex != null) // Noncompliant FP
+            {
+            }
+        }
+
+        public void ElementAccess2(Exception ex)
+        {
+            if (ex?.Data["Key"] is string value)
+            {
+            }
+            else if (ex.Data != null) // FN
+            {
+            }
+        }
+
+        public void MemberBinding1(Exception ex)
+        {
+            if (ex?.Message is string value)
+            {
+            }
+            else if (ex != null) // Noncompliant FP
+            {
+            }
+        }
+
+        public void MemberBinding2(Exception ex)
+        {
+            if (ex?.Message is string value)
+            {
+            }
+            else if (ex.Message != null) // FN
```

Comment:
on the `else` branch there can be 3 cases:
1. `ex` is null
2. `ex` is not null, `ex.Message` is  null
3. `ex` is not null, `ex.Message` is  not null and not string

Given that `Message` may be either null or not null, a null check makes sense

Reply:
`Message` is of type string, therefore `not string` is not a concern.
If you enter `ex?.Message is string value` branch, `Message` is definitely not null. 
In else branch, `ex` might be null, resulting in null reference exception. But `ex.Message` is definitely null then, therefore FN.

I'll add the `ex == null` check.